sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -569,6 +575,40 @@ public static void SetTraceState(this Activity activity, string? tracestate)
             SetTraceStateStringMethod(activity, tracestate);
         }
 
+        public static void SetErrorStatus(this Activity activity, string description)
+        {
+            if (SetErrorStatusMethod == null)
+            {
+                if (ActivityStatusCodeType == null)
+                {
+                    SetErrorStatusMethod = (_, _, _) => { };
+                }
+                else
+                {
+                    var method = typeof(Activity).GetMethod("SetStatus", BindingFlags.Instance | BindingFlags.Public, null, new Type[]
+                    {
+                        ActivityStatusCodeType,
+                        typeof(string)
+                    }, null);
+                    if (method == null)
+                    {
+                        SetErrorStatusMethod = (_, _, _) => { };
+                    }
+                    else
+                    {
+                        var methodParameters = method.GetParameters();
+
+                        var statusParameter = Expression.Parameter(typeof(int));
+                        var descriptionParameter = Expression.Parameter(typeof(string));
+                        SetErrorStatusMethod = Expression.Lambda<Action<Activity, int, string>>(
+                            Expression.Call(ActivityParameter, method, Expression.Convert(statusParameter, methodParameters[0].ParameterType), descriptionParameter),
```

Comment:
Would it make sense to cache these lambdas?

Reply:
We do cache these.

Reply:
Oh, I see. `SetErrorStatusMethod` persists for future invocations.